mindmeld/components/role_classifier.py

```diff
@@ -313,7 +317,7 @@ def _get_queries_and_labels(self, queries=None, label_set=DEFAULT_TRAIN_SET_REGE
                     labels.append(entity.entity.role)
 
         unique_labels = set(labels)
-        if len(unique_labels) == 1:
+        if len(unique_labels) == 0:
```

Comment:
what is the `0` case? I thought unlabeled roles count as `label: None`, and just aren't printed. so if there are any entities, there would be some role labels in this list, right?

Reply:
Let me check but I assume that currently `label: None` is actually a case of bad annotation (line 323). So `unique_labels` refers to the set of all valid Role labels:
```
if None in unique_labels:
            bad_examples = [e for i, e in enumerate(examples) if labels[i] is None]	            bad_examples = [e for i, e in enumerate(examples) if labels[i] is None]
```

Reply:
Does it not distinguish between entities that have a role vs entities that don't? e.g. `{8:00|sys_time|start_time}` vs `{December|sys_time}`

Reply:
![image](https://user-images.githubusercontent.com/1332283/93649934-03be5b80-f9c2-11ea-9f39-ca2d76a4a802.png)
I think you can test it again locally, but apparently the current role classifier does not distinguish between entities that have a role and entities that don't. Here in this query ("change my alarm from 6am to 8am since it is already 3am") all three entities are annotated with a role type even though only two of them were marked so in training and test.